diff --git a/include/openssl/opensslv.h b/include/openssl/opensslv.h
index b866670e18..ae7b7f3ade 100644
--- a/include/openssl/opensslv.h
+++ b/include/openssl/opensslv.h
@@ -40,7 +40,7 @@ extern "C" {
  *  major minor fix final patch/beta)
  */
 # define OPENSSL_VERSION_NUMBER  0x1010100aL
-# define OPENSSL_VERSION_TEXT    "OpenSSL 1.1.1-pre10-dev  xx XXX xxxx"
+# define OPENSSL_VERSION_TEXT    "OpenSSL 1.1.1-pre10-dev  10 Sep 2018"
 
 /*-
  * The macros below are to be used for shared library (.so, .dll, ...)
